### PR TITLE
Osd 19746 svclog

### PR DESCRIPTION
**Pull request by kamlendu1982**
Acceptance criteria:

    osdctl will check if another (non-automated) servicelog has been sent to the cluster already within the last hour.
    If another servicelog has been sent already it will as the user to confirm sending the additonal servicelog.
    If osdctl is not running in an interactive context this check is disabled.

### CODE REVIEW

File: cmd/servicelog/post.go
Comment (typeid) [inline cmd/servicelog/post.go:109]: In case there was at least one service log, we probably want to print it/them as it is necessary information for the decision whether a new SL should be sent or not.

File: cmd/servicelog/post.go
Comment (typeid) [inline cmd/servicelog/post.go:121]: I believe we already have a helper function that could be re-used here: https://github.com/openshift/osdctl/blob/cdfaf84d3593afed1797b42758f657baebdf04aa/pkg/utils/utils.go#L249-L264

File: cmd/servicelog/post.go
Comment (typeid) [inline cmd/servicelog/post.go:104]: nitpick: this comment has nothing to do here, the `CheckServiceLogsLastHour()` logic doesn't know what's happening outside of it.

**Review by typeid — CHANGES_REQUESTED** [review]

**Comment by typeid** [comment]
Thanks for the contribution @kamlendu1982! I added some small comments, but generally this is already very close to the goal :)

File: cmd/servicelog/post.go
Comment (typeid) [inline cmd/servicelog/post.go:104]: I think it would be cleaner to print this line just once in case we had service logs in the past hour, then print the list of service logs - but I'll leave this up to you whether you want to change this or not.

File: cmd/servicelog/post.go
Comment (typeid) [inline cmd/servicelog/post.go:92]: nitpick: this could be private

```suggestion
func (o *PostCmdOptions) checkServiceLogsLastHour() bool {
```

File: cmd/servicelog/post.go
Comment (typeid) [reply cmd/servicelog/post.go:95]: +1 Always good to have things configurable. :)

File: cmd/servicelog/post.go
Comment (typeid) [inline cmd/servicelog/post.go:104]: Typo:
```suggestion
			fmt.Println("Below service log has been submitted in last 60 minutes\nDescription: ", svclog.Description())
```

**Comment by kamlendu1982** [comment]
incorporated the recommend changes and addressed the review comments.
Can you please have a final review.

File: cmd/servicelog/post.go
Comment (bergmannf) [inline cmd/servicelog/post.go:117]: Had a talk today about another card that would add a check like this.
It might be useful to still print that there was a warning even if it doesn't stop the script (e.g. to STDERR) just so it can be seen.

**Comment by T0MASD** [comment]
tide/merge-method-squash

**Comment by typeid** [comment]
/approve

@kamlendu1982 could you please mark the comments as resolved if they no longer apply?

**Pull request closed without merging**
